src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -79,6 +80,17 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(defn static-file-content-request-handler
+  "Build the main request handler for the /static_file_content endpoint"
+  [get-code-content]
+  (fn [req]
+    (let [environment (get-in req [:params "environment"])
+          code-id (get-in req [:params "code-id"])
+          file-path (get-in req [:params :rest])
+          content-stream (get-code-content environment code-id file-path)]
+      {:status 200
```

Comment:
Yup. Not only that, but even if we could check the file inside the environment, it's entirely possible that the file exists at the revision indicated by `code-id` but was deleted in a later revision, and thus does not exist in the most recent code.